l5kit/l5kit/environment/callbacks.py

```diff
@@ -0,0 +1,75 @@
+from typing import List
+
+import gym
+from stable_baselines3.common.callbacks import EvalCallback
+
+from l5kit.cle.validators import ValidationCountingAggregator
+from l5kit.environment.envs.l5_env import EpisodeOutputGym
+from l5kit.environment.gym_metric_set import DisplacementCollisionMetricSet
+
+
+class L5KitEvalCallback(EvalCallback):
+    """Callback for evaluating an agent using L5Kit evaluation metrics.
+
+    :param eval_env: The environment used for initialization
+    :param n_eval_episodes: The number of episodes to test the agent
+    :param eval_freq: Evaluate the agent every ``eval_freq`` call of the callback.
+    :param verbose:
+    :param metric_set: computes a set of metric parametrization for the L5Kit environment
+    """
+
+    def __init__(self, eval_env: gym.Env, eval_freq: int = 10000, n_eval_episodes: int = 10,
+                 n_eval_envs: int = 4, verbose: int = 0) -> None:
+        super(L5KitEvalCallback, self).__init__(eval_env)
+        self.eval_freq = eval_freq
+        self.n_eval_episodes = n_eval_episodes
+        self.n_eval_envs = n_eval_envs
+        self.verbose = verbose
+        self.metric_set = DisplacementCollisionMetricSet()
+
+    def _init_callback(self) -> None:
+        pass
+
+    def _on_step(self) -> bool:
+        if self.eval_freq > 0 and self.n_calls % self.eval_freq == 0:
+            # Get episode outputs
+            episode_outputs = self.get_eval_info()
+
+            # Evaluate
+            self.metric_set.evaluator.evaluate(episode_outputs)
+            validation_results = self.metric_set.evaluator.validation_results()
+            _ = ValidationCountingAggregator().aggregate(validation_results)
+
+            # TODO Log the aggregated results
```

Comment:
You can add all metrics we have available for the CLE in l5kit.

Reply:
I wanted to ask what is usually the case? e.g.
1. All the `metrics`? Averaged across time or measured at 'last' time step?
2. All the `range validators`?
3. Both?

Reply:
We already have the `ValidationCountingAggregator` there, so you can get from this aggregation (it will aggregate per scene), at the end of the day we want to know for the entire number of scenes, what were the results aggregated (how many collisions in total, for the entire amount of scenes, for a particular policy)

Reply:
Done!